packages/extensions-sdk/src/cli/commands/validators/check-directus-config.ts

```diff
@@ -0,0 +1,115 @@
+import { EXTENSION_PKG_KEY, EXTENSION_TYPES, API_EXTENSION_TYPES } from '@directus/extensions';
+import fse from 'fs-extra';
+import { type Ora } from 'ora';
+import type { Report } from '../../types.js';
+import { isObject } from '@directus/utils';
+
+const checkDirectusConfig = {
+	name: 'directus-config',
+	handler: async (spinner: Ora, reports: Array<Report>) => {
+		spinner.text = 'Checking package file exists';
+
+		if (!(await fse.pathExists(`${process.cwd()}/package.json`))) {
+			spinner.fail();
+
+			const message = 'No package.json';
+
+			reports.push({
+				level: 'error',
+				message: `${checkDirectusConfig.name}: ${message}`,
+			});
+
+			throw new Error(message);
+		}
+
+		const packageFile = await fse.readJson(`${process.cwd()}/package.json`);
+
+		spinner.text = `Checking ${EXTENSION_PKG_KEY} is present`;
+
+		if (!packageFile[EXTENSION_PKG_KEY]) {
+			spinner.fail();
+
+			const message = 'Directus object not found';
+
+			reports.push({
+				level: 'error',
+				message: `${checkDirectusConfig.name}: ${message}`,
+			});
+
+			throw new Error(message);
+		}
+
+		const { type, host, sandbox } = packageFile[EXTENSION_PKG_KEY];
+		let { path } = packageFile[EXTENSION_PKG_KEY];
+
+		spinner.text = `Checking extension type`;
+
+		if (!EXTENSION_TYPES.includes(type)) {
+			spinner.fail();
+			const message = `Invalid Directus Extension Type: ${type}`;
+
+			reports.push({
+				level: 'error',
+				message: `${checkDirectusConfig.name}: ${message}`,
+			});
+
+			throw new Error(message);
+		}
+
+		spinner.text = `Checking extension path(s)`;
+
+		if (!isObject(path)) {
+			path = { app: path };
+		}
+
+		Object.keys(path).forEach(async (key) => {
+			if (!(await fse.pathExists(`${process.cwd()}/${path[key]}`))) {
+				spinner.fail();
+				const message = `Extension path ${key}: ${path[key]} invalid`;
+
+				reports.push({
+					level: 'error',
+					message: `${checkDirectusConfig.name}: ${message}`,
+				});
+
+				throw new Error(message);
+			}
+		});
+
+		spinner.text = 'Checking for valid Directus host version';
+
+		const regex = new RegExp(/^\^?[1-9]\d*(\.[1-9]\d*)*/);
```

Comment:
I think we document that `host` has to be semver, so let's check for a valid semver (three parts) instead of any version number looking string.

https://docs.directus.io/extensions/creating-extensions.html#extension-folder-structure

Reply:
I'd noticed some extensions that have already been published using the pipe separated list of version (eg `^10.1.3 || 11.0.0` ). Do we want to validate for those as well or just a single semver?

Reply:
Interesting! Under the hood (for display on the marketplace page) we use the [`semver`](https://www.npmjs.com/package/semver) package, which comes with `validRange` we could use for testing the `host` version.